### ISSUE TITLE
Bintray account for Streamlink has been rate-limited

### ISSUE THREAD

**Issue by bastimeyer** [streamlink/streamlink#1476]
> Hi Streamlink,
> 
> Your Bintray account is currently rate-limited due to overuse.
> 
> Customer Support should contact you at the earliest to remove the limits from your account. You may also contact them directly at support@jfrog.com.
> 
> Regards,
> The Bintray Team

I don't know whether the downloads have been abused or not (it doesn't let me see any download statistics on their website), but it looks like we've reached the traffic limit for this month already, and it's just the 9th of February today.

Maybe it's time thinking about a different release model that doesn't force Windows users to download nightlies from an external, auxiliary hoster, just because they want the latest version of a specific plugin. For now, we should add a small note to the download page to tell users to consider downloading a stable release instead of a nightly build, unless they've experienced a bug in the latest version.

As we've already discussed a few months ago (#811), using Github as a hoster for pre-releases could work, but only if we set up a different repo, so that the amount of notifications and spam can be kept down. I don't know though if we would be breaking Github's ToS with that.

Regarding the idea of letting Streamlink update its plugins automatically (#1347), I've already said that I don't think that we should turn Streamlink into a package manager of its own, as there are too many things which need to be taken into account and which also can go wrong. I also doubt that anybody would want to implement this kind of feature here.

My suggestion in order to reduce the number of nightly downloads, is publishing releases more often, in a 14-day schedule or so. This could even be automated, if desired. I don't actively use yt-dl, but they're releasing far more often and are using the [calendar versioning scheme](https://calver.org/#youtube_dl). This lets people immediately know how old a release is and if we'd publish a new release every two weeks, the intention of downloading a nightly build gets reduced greatly.

**Comment by gravyboat** [comment]
I got this notification as well. I wish bintray had more stats. It seems odd to me that so many people would download for the limited number of plugins that get updated where we would be over for the month, but without more info there's not much we can do there.

I agree that it seems like the best strategy going forward is releasing more regularly. I still don't really like the idea of doing pre-releases on GitHub and have no idea how abuse would work in that situation. Adding a note to mention that users should only download the nightly if something is wrong, as well as not noting it quite as prominently on the docs might be a good option. As long as we aren't charged I'm not too concerned about hitting the rate limit either, but I'm also biased.

**Comment by amurzeau** [comment]
statistics about downloads/day or not available / sufficient ? Like that: https://bintray.com/amurzeau/streamlink-debian/streamlink#statistics

**Comment by bastimeyer** [comment]
The last build seems to have failed during the upload of the Windows installers to Bintray: 
https://travis-ci.org/streamlink/streamlink/jobs/344938894#L3346
Let's see if this keeps happening...

**Comment by thinkpad4** [comment]
With Bintray rate limited where can I get the latest nightly from?

**Comment by beardypig** [comment]
@thinkpad4 currently we don't have a solution for nightly builds. It is possible to install python + pip and use the latest version from github (https://streamlink.github.io/install.html#source-code).

**Comment by RosadinTV** [comment]
@thinkpad4 
You can try the [Portable Builder](https://github.com/streamlink/streamlink-portable) (requires Win7 or later, but maybe works with Vista SP2 too).

**Comment by thinkpad4** [comment]
OK, thanks

**Comment by thinkpad4** [comment]
Still no solution?

**Comment by gravyboat** [comment]
@thinkpad4 Not past what beardypig suggested. If you don't mind my asking why are you using the nightly builds so much?

**Comment by thinkpad4** [comment]
I'm not like installing a new nightly every day, it is just after I see a fix for a plugin that I use get added I wanna install it instead of waiting for the next main release

**Comment by bastimeyer** [comment]
Bintray is becoming more and more unreliable. Every second or third cron job is failing during the bintray upload:
https://travis-ci.org/streamlink/streamlink/jobs/360401930#L2672
https://travis-ci.org/streamlink/streamlink/builds

**Comment by back-to** [comment]
> Your Bintray account is currently rate-limited due to overuse.

I think it's **not about traffic**, it's more about a **full storage**.

```
What's included in the OSS?

    Universal Support for All Package Types
    1 TB Downloads
    10 GB Storage
    CDN
```

from https://bintray.com/signup/oss

---

the last version is **2018.02.09**

https://bintray.com/streamlink/streamlink-nightly/streamlink

> 275
> Days between 10 May 2017 - 09 Feb 2018 https://github.com/streamlink/streamlink/pull/897

let's say **~36 MB** per exe file

36MB * 275 = 9900 MB

you are at/over 10 GB

---

is there a way to delete the old files?

**Comment by beardypig** [comment]
I have deleted all the old versions, but the upload is still failing. I guess the next step is to contact them.

**Comment by beardypig** [comment]
Removing the old versions seems to have fixed it now. I will be updating the deploy script to automatically remove versions older than 30 days so we don't run in to the 10GB limit again (thanks @back-to...)

@thinkpad4 nightlies are back.

**Comment by thinkpad4** [comment]
@beardypig YAY, thank you kindly

**Comment by bastimeyer** [comment]
I think we should remove the `streamlink-latest.exe` download option. This can sometimes be confusing, like in this case: https://github.com/streamlink/streamlink/issues/1615#issuecomment-383702485
Since [we're not linking directly to the download anymore](https://github.com/streamlink/streamlink/pull/1408/files#diff-62c330cd8da31d476bb63b51bb8b1307L253), I think this shouldn't cause any issues...
@beardypig could you please keep that in mind when you're going to change the bintray deploy script? Thanks!

**Comment by beardypig** [comment]
I already made that change, months ago even ... but never pushed it ;( I'll put together a PR with the fixes today :)

**Comment by beardypig** [comment]
This should be resolved as of #1624 :)
